pkg/controller/stackconfigpolicy/controller.go

```diff
@@ -291,6 +308,22 @@ func (r *ReconcileStackConfigPolicy) doReconcile(ctx context.Context, policy pol
 			return results.WithError(err), status
 		}
 
+		// create secrets that are present in the secret mounts
+		err = reconcileSecretMountSecretsESNamespace(ctx, r.Client, es, &policy)
+		if err != nil {
+			return results.WithError(err), status
+		}
+
+		// create expected elasticsearch config secret
+		expectedConfigSecret, err := NewElasticsearchConfigSecret(policy, es)
+		if err != nil {
+			return results.WithError(err), status
+		}
+
+		if err := ReconcileElasticsearchConfigSecret(ctx, r.Client, expectedConfigSecret, es, policy); err != nil {
+			return results.WithError(err), status
+		}
+
 		// get /_cluster/state to get the Settings currently configured in ES
 		currentSettings, err := r.getClusterStateFileSettings(ctx, es)
```

Comment:
Maybe keep this part near line `351` as `currentSettings` is not used until there.